enhancements/installer/openstack-customer-provided-subnets.md

```diff
@@ -0,0 +1,131 @@
+---
+title: allow-customer-provisioned-aws-subnets
+authors:
+  - "@wking"
+reviewers:
+  - "@mandre"
+  - "@pierreprinetti"
+  - "@adduarte"
+approvers:
+  - "@abhinavdahiya"
+  - "@sdodson"
+creation-date: 2020-03-12
+last-updated: 2020-03-12
+status: planning
+---
+
+# 4.5: OpenStack: Allow Customer Provisioned Subnets
+
+## Release Signoff Checklist
+
+- [ x ] Enhancement is `implementable`
+- [ x ] Design details are appropriately documented from clear requirements
+- [ x ] Test plan is defined
+- [ x ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+In many large organizations, IT groups are separated into teams. A common separation of duties lies between networking/WAN and application or enabling technology deployments. One group will provision or assign OpenStack accounts with core management technology. A second group will be responsible for provisioning networks complete with subnet & zone selection, routing, VPN connections, and ingress/egress rules from the company. Lastly, an applications team (or individual business units) are given a quota for other OpenStack resources to provision application-specific items (instances, containers, databases, pipelines, load balancers, etc.).
```

Comment:
> ingress/egress rules from the company

are ingress/egress firewall rules network level in Openstack?

Reply:
My understanding is that they are enforced via Linux IP table rules on the nodes. However, that may be outdated.

Reply:
The standard way of enforcing ingres/egress rules in OpenStack is via "security groups" -- rules in the OpenStack SDN (Neutron). One or more groups can be attached to an OpenStack server (VM) and the filtering happens on the Neutron level. So they're set per node but the enforcement is done in the OpenStack networking piece.

This happens a layer above the node/VM and it is completely separate from the operating system firewall.

Reply:
Thanks Tomas! In that case, we should not configure security groups for user provisioned subnets.

Reply:
Just as a technnical  reference. in openstack the security rules are implemetned either near the vm in the same openstack compute node by addin rules to a linux bridge, or by implementing openflow rules on an openvswitch switch, again in the same openstack compute node.  so in short, the seucrity groups get translated into rues which get programmed into either a linux bridge or an openvswitch bridge, depending on the driver that has been selected.

Reply:
There has been a lot of debate and discussion about this, and ultimately it seems like the installer should not have a problem creating security groups and rules.